daemon/volumes.go

```diff
@@ -226,12 +226,15 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 			bind.Volume = v
 			bind.Source = v.Path()
 			// Since this is just a named volume and not a typical bind, set to shared mode `z`
-			if bind.Relabel == "" {
-				bind.Relabel = "z"
+			if bind.Mode == "" {
+				bind.Mode = "z"
 			}
 		}
 
-		if err := label.Relabel(bind.Source, container.MountLabel, bind.Relabel); err != nil {
+		if err := label.Relabel(bind.Source, container.MountLabel, bind.Mode); err != nil {
+			return err
+		}
+		if err := volume.FixUidGid(container.Config.User, bind.Source, bind.Mode); err != nil {
```

Comment:
Why it will be dealing with uid? `container.Config.User` is a string and it can specify username, gid, groupname and stuff.